ibllib/io/video.py

```diff
@@ -31,7 +30,7 @@ def get_video_frame(video_path, frame_number):
 
 
 def get_video_frames_preload(video_path, frame_numbers=None, mask=Ellipsis, as_list=False,
-                             func=lambda x: x,):
+                             func=lambda x: x, cap=None):
```

Comment:
It doesn't make sense to provide both a video path and and a VideoCapture class, especially as the video_path is a positional argument.  Instead perhaps the first input can be one of (cv2.VideoCapture, VideoStreamer, str, Path)?

Reply:
Ah yes good point. I assume the parameter name cannot be changed though, right? (As in I cannot call it video_path_or_cap to be more explicit)?
I will adapt the code and try to make it clear in the docstring then

Reply:
You can rename the positional input variables to what ever you like, it won't break anyone's code.  I suggested 'vid' because it encompasses any representation of a video, including the VideoStreamer object, but it's up to you - it's not that important.